akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -61,7 +64,21 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
         props.args map serialize foreach builder.addArgs
-        props.args map (a ⇒ if (a == null) "null" else a.getClass.getName) foreach builder.addClasses
+        props.args.map { a ⇒
+          val argClassName =
+            if (a == null) "null"
+            else {
+              val className = a.getClass.getName
+              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && className.contains("$Lambda$")) {
+                // The serialization of the parameters is based on passing class name instead of
```

Comment:
> Instead of passing the lamda class name I pass "java.io.Serializable".

This is the part that I could not understand. How can this possibly work? Because the class is actually existing it is just created via the LMF?